src/Symfony/Component/Config/Loader/FileLoader.php

```diff
@@ -100,9 +110,11 @@ public function import($resource, $type = null, $ignoreErrors = false, $sourceRe
             self::$loading[$resource] = true;
 
             try {
+                $this->setCurrentResource($resource);
                 $ret = $loader->load($resource, $type);
             } finally {
                 unset(self::$loading[$resource]);
+                $this->setCurrentResource($currentResource);
```

Comment:
this is wrong, as it does not reset the object in the previous state (as ``$currentResource`` is not set directly in the property here, and may not come from the property either).

Adding 1 more state property in the loader is a bad idea IMO, as state in service objects makes them harder to use.